### PR TITLE
Gci 907 delivery details page cert copies

### PR DESCRIPTION
**Pull request by ethatcher83**
extract delivery details validation to common file
refactor certificates delivery details controller to use common validation
create certs copies controller
change back link on delivery details template to be dynamic based on the service user is in
extract delivery details template as now a common file
add delivery details integration test

Resolves: GCI-907

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -70,9 +70,11 @@ app.use((req, res, next) => {
     if (req.path.includes("/certificates")) {
         env.addGlobal("SERVICE_NAME", SERVICE_NAME_CERTIFICATES);
         env.addGlobal("SERVICE_PATH", "/certificates");
+        env.addGlobal("DELIVERY_DETAILS_BACK_LINK", "certificate-options");
```
Comment (nabeelamirch) [inline src/app.ts:73]: This variable is used only on the two delivery details page. So I would put them in both controllers and not here.

File: src/controllers/certified-copies/delivery.details.controller.ts [hunk 2]
```diff
@@ -1,16 +1,74 @@
...
 
 const logger = createLogger(APPLICATION_NAME);
 
 export const render = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-        res.status(200).json({ value: true });
+        const userId = getUserId(req.session);
+        const accessToken: string = getAccessToken(req.session);
+        const basket: Basket = await getBasket(accessToken);
+        // logger.info(`Get certificate item, id=${certificateItem.id}, user_id=${userId}, company_number=${certificateItem.companyNumber}`);
```
Comment (nabeelamirch) [inline src/controllers/certified-copies/delivery.details.controller.ts:27]: Keep a comment here to say that the basket is being fetched for a user with the id

File: src/controllers/certified-copies/delivery.details.controller.ts [hunk 3]
```diff
@@ -1,16 +1,74 @@
...
+        return res.render(CERTIFIED_COPY_DELIVERY_DETAILS, {
```
Comment (nabeelamirch) [inline src/controllers/certified-copies/delivery.details.controller.ts:28]: Call this variable DELIVERY_DETAILS as they both have the same value.

File: src/model/template.paths.ts [hunk 4]
```diff
@@ -7,10 +7,10 @@ export const COLLECTION: string = "collection";
 export const CERTIFICATE_INDEX: string = `${certRoot}/index`;
 export const CERTIFICATE_OPTIONS: string = `${certRoot}/options`;
 export const CERTIFICATE_CHECK_DETAILS: string = `${certRoot}/check-details`;
-export const CERTIFICATE_DELIVERY_DETAILS: string = `${certRoot}/delivery-details`;
+export const CERTIFICATE_DELIVERY_DETAILS: string = "delivery-details";
```
Comment (nabeelamirch) [inline src/model/template.paths.ts:10]: Call this variable DELIVERY_DETAILS as they both have the same value.

File: src/app.ts (hunk 1 above)
Comment (ethatcher83) [reply src/app.ts:73]: updated, thanks

File: src/model/template.paths.ts (hunk 4 above)
Comment (ethatcher83) [reply src/model/template.paths.ts]: updated, thanks

File: src/controllers/certified-copies/delivery.details.controller.ts (hunk 2 above)
Comment (ethatcher83) [reply src/controllers/certified-copies/delivery.details.controller.ts]: updated, thanks

File: src/app.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethatcher83) [reply src/app.ts]: resolved

**Pull request merged**